index.js

```diff
@@ -1,49 +1,70 @@
+/* eslint-disable no-console */
+
 "use strict";
 
 const connoisseur = require("./lib/connoisseur");
 const fs = require("fs");
 const concat = require("concat-stream");
-const sourceFiles = require("yargs").argv._;
-
-const useStdIn = process.argv.indexOf("--stdin") > -1;
+const argv = require("yargs").boolean("f").argv;
+const useFile = argv.f;
 
 const fixIndentation = code => code.replace(/(\r\n|\n|\r)/gm, "\n\t");
 
 const displayFileErrors = function (fileInfo) {
-    if (!fileInfo.errors)
-        return;
-
-    /* eslint-disable no-console */
     console.log(
         `Errors found for file ${fileInfo.path}:\n${
             fileInfo.errors.reduce((prev, errorData) =>
-                    `${prev}\t${fixIndentation(errorData.code)}\t${
-                        errorData.error || `→ Expected: ${errorData.expected}, Found: ${errorData.actual}`
-                    }\n`
-                , "")
-            }`
+                `${prev}\t${fixIndentation(errorData.code)}\t${
+                    errorData.error || `→ Expected: ${errorData.expected}, Found: ${errorData.actual}`
+                }\n`
+            , "")
+        }`
     );
-    /* eslint-enable no-console */
 };
 
-if (useStdIn) {
+if (useFile) {
+    const sourceFiles = argv._;
+
+    const files = sourceFiles.filter(filePath => {
+        if (fs.existsSync(filePath))
+            return true;
+
+        console.log(`File '${filePath}' doesn't exist.`);
+
+        return false;
+    });
+
+    if (!files.length) {
+        console.log("No files found.");
+        process.exitCode = 1;
+    }
+
+    const errors = files.map(filePath => ({
+        path: filePath,
+        errors: connoisseur(fs.readFileSync(filePath, "utf8")),
+    }))
+    .filter(fileInfo => fileInfo.errors.length);
+
+    if (errors.length) {
+        errors.forEach(displayFileErrors);
+        process.exitCode = 1;
+    } else {
+        console.log("No errors found.");
+        process.exitCode = 0;
+    }
+} else {
     process.stdin.pipe(concat({ encoding: "string" }, text => {
```

Comment:
I'd try normalizing this as a "normal" file with the name `STDIN` and use the previous path (the non stdin path with multiple file names) if possible in a follow-up.